internal/model/task.go

```diff
@@ -147,3 +148,48 @@ func (t *Task) SetState(state sw.State) error {
 func (t *Task) State() sw.State {
 	return sw.State(t.state)
 }
+
+func NewTaskStatusRecord(s string) StatusRecord {
+	sr := StatusRecord{}
+	if s == "" {
+		return sr
+	}
+
+	sr.Append(s)
+
+	return sr
+}
+
+type StatusRecord struct {
+	StatusMsgs []StatusMsg `json:"records"`
+}
+
+type StatusMsg struct {
+	Timestamp time.Time `json:"ts,omitempty"`
+	Msg       string    `json:"msg,omitempty"`
+}
+
+func (sr *StatusRecord) Append(s string) {
+	if s == "" {
+		return
+	}
+
+	for _, r := range sr.StatusMsgs {
```

Comment:
Thats a good point, although then the records have to be sorted by time before being displayed to the user and that makes it a slight bit more cumbersome, since this `StatusMsg` type is flasher type - I guess it could be moved into rivets if other controllers require something as such.. :ponder